migrations/lib/importer/steps/category_upload_references.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Migrations::Importer::Steps
+  class CategoryUploadReferences < ::Migrations::Importer::Step
+    depends_on :categories
+
+    def execute
+      super
+
+      DB.exec(<<~SQL)
+        INSERT INTO upload_references (upload_id, target_type, target_id, created_at, updated_at)
+        SELECT upload_id, 'Category', target_id, created_at, updated_at
+        FROM (
+               SELECT uploaded_logo_id AS upload_id, id AS target_id, created_at, updated_at
+                 FROM categories
+                WHERE uploaded_logo_id IS NOT NULL
+                UNION ALL
+               SELECT uploaded_logo_dark_id AS upload_id, id AS target_id, created_at, updated_at
+                 FROM categories
+                WHERE uploaded_logo_dark_id IS NOT NULL
+                UNION ALL
+               SELECT uploaded_background_id AS upload_id, id AS target_id, created_at, updated_at
+                 FROM categories
+                WHERE uploaded_background_id IS NOT NULL
+                UNION ALL
+               SELECT uploaded_background_dark_id AS upload_id, id AS target_id, created_at, updated_at
+                 FROM categories
+                WHERE uploaded_background_dark_id IS NOT NULL
+             ) AS category_upload_refs
+        WHERE NOT EXISTS (
+                           SELECT 1
+                             FROM upload_references ur
+                            WHERE ur.upload_id = category_upload_refs.upload_id
+                              AND ur.target_type = 'Category'
+                              AND ur.target_id = category_upload_refs.target_id
+                         )
```

Comment:
I stopped using the `WHERE NOT EXISTS` for inserting into the `upload_references` table. The `ON CONFLICT DO NOTHING` does the job well and during my work on user avatars, I noticed that without it, it was a lot faster. Like 1 second vs half a minute.

Reply:
I was considering that too, so I ran some `EXPLAIN` tests with and without it and also switched from `UNION` to `UNION ALL`. And yes, on my limited test data, it was generally faster without the `WHERE NOT EXISTS` check, though, a lot of the difference came from planning time. I left it in because, I figured with enough records, planning time won't matter much. Then again, categories usually aren’t that large, so in practice we might not be dealing with a dataset big enough for this to matter. I'm going to remove it for now.

That said, it got me thinking about resumability. I don’t have enough data to test this properly, but have you tried a scenario where more than half the references already exist? In that case, does relying solely on `ON CONFLICT DO NOTHING` still perform better?